### PR TITLE
Support Reason 3 and Next.js 4

### PR DESCRIPTION
**Pull request by dcalhoun**
Convert source to [Reason 3](https://reasonml.github.io/community/blog/#reason-3) and bump version to [Next.js 4](https://zeit.co/blog/next4).

### Additional Changes

The following were done with the primary goal of aligning with the [ReasonReact repo structure](https://github.com/reasonml/reason-react) and following the recommendations found in the ReasonML docs.

* Capitalize Reason source files. ([ref](https://reasonml.github.io/guide/meta/project-structure#file-casing))
* Include JS output in version control. ([ref](https://reasonml.github.io/guide/javascript/converting#tips)) ([ref](https://github.com/reasonml/reason-react/tree/master/lib/js/src))
* Utilize Yarn for tracking package versions.
* Add documentation and examples.

---

I am _really_ new to ReasonML. So, please double check my conversion. I basically stumbled through the syntax until it began working for my app. 😊

### CODE REVIEW

**Comment by ulrikstrid** [comment]
Hey, I've been sick for a while so have some catching up to do, but I will have a look at this during the day!

Thanks!

File: src/Next.re [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            "passHref": from_opt(passHref)
+          }
+        ),
+      children
+    );
+};
+
+module Head = {
+  [@bs.module "next/head"] external head : ReasonReact.reactClass = "default";
+  let make = (_children) =>
```
Comment (ulrikstrid) [inline src/Next.re]: `children` should not be underscored as that indicates that it isn't  used.

**Review by ulrikstrid — CHANGES_REQUESTED** [review]
Looks good, just one small tweak.

**Comment by dcalhoun** [comment]
@ulrikstrid Thank you for taking the time to review. I've addressed your feedback.

**Comment by ulrikstrid** [comment]
@dcalhoun thanks a lot for this! 🥇

I will get this up on npm soon, I don't have a maching with a working reason setup at the moment.

**Comment by ulrikstrid** [comment]
This is finally up, sorry for being slow

**Pull request merged**